pkg/webhook/resources/upgrade/validator.go

```diff
@@ -514,3 +524,39 @@ func (v *upgradeValidator) getKubeletStatsSummary(nodeName, kubeletURL string) (
 	}
 	return summary, nil
 }
+
+func (v *upgradeValidator) checkCerts(version *v1beta1.Version) error {
+	kubernetesIP, err := util.GetKubernetesIP(v.serviceCache)
+	if err != nil {
+		return werror.NewInternalError(fmt.Sprintf("can't get kubernetes ip, err: %+v", err))
+	}
+	if kubernetesIP == "" {
+		err = fmt.Errorf("cluster ip is empty")
+		logrus.WithFields(logrus.Fields{
+			"service.namespace": metav1.NamespaceDefault,
+			"service.name":      "kubernetes",
+		}).WithError(err).Error("cluster ip is empty in the service")
+		return werror.NewInternalError(fmt.Sprintf("can't get kubernetes ip, err: %+v", err))
+	}
+
+	earliestExpiringCert, err := util.GetEarliestExpiringCert(fmt.Sprintf("%s:443", kubernetesIP))
+	if err != nil {
+		return werror.NewInternalError(fmt.Sprintf("can't get earliest expiring cert, err: %+v", err))
+	}
+
+	minCertsExpirationInDayString := "7"
```

Comment:
If I read the code correctly, this is the default minimum time until expiration. This should be a constant and not an ad-hoc definition.